### PR TITLE
Move codebase to Node 4

### PR DESCRIPTION
**Pull request by ooogway**
not the best time for a PR.. merry xmas guys and a very happy new year.

not sure if it was planned, but this PR moves the existing code base to Node 4 and drop support for previous version of node.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: node_js
 
 node_js:
-  - 0.10
```
Comment (Marsup) [inline .travis.yml]: Inspiration : https://github.com/hapijs/hapi/blob/master/.travis.yml

File: bin/confidence [hunk 2]
```diff
@@ -30,8 +31,8 @@ internals.argv = Yargs.usage('Usage: $0 -c config.json [--filter.criterion=value
 
 internals.getConfig = function () {
 
-    var config = null;
-    var configPath = internals.argv.c[0] !== '/' ? process.cwd() + '/' + internals.argv.c : internals.argv.c;
+    let config = null;
+    const configPath = internals.argv.c[0] !== '/' ? process.cwd() + '/' + internals.argv.c : internals.argv.c;
```
Comment (Marsup) [inline bin/confidence]: Could make use of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

**Comment by ooogway** [comment]
@Marsup added suggested changes

**Comment by gergoerdosi** [comment]
Any reason this is not merged yet?

**Comment by gergoerdosi** [comment]
@patrickkettner: We rely on this module in many of our projects. We would like to make sure it is maintained and works with all the recent node versions. If anything blocks these changes from being merged, please let me know and I will send an other pull request.

**Comment by lloydbenson** [comment]
@gergoerdosi I was able to get a hold of @patrickkettner and he said he should be able to take care of this soon (tuesday).  He has just been busy with a newborn ;)

**Comment by gergoerdosi** [comment]
Thanks for the information @lloydbenson. Congrats @patrickkettner!

**Comment by patrickkettner** [comment]
Sorry again, everyone. Diving into this now

File: .travis.yml (hunk 1 above)
Comment (patrickkettner) [inline .travis.yml]: could this be updated to follow what hapi is [doing now](https://github.com/hapijs/hapi/commit/e42ed26be1784977a2fd0706136c08eb413da5b7)?

**Comment by patrickkettner** [comment]
just one tiny change, and then we can merge and release. Sorry again for this taking forever

**Comment by ooogway** [comment]
@patrickkettner : added suggested change

**Comment by patrickkettner** [comment]
published as 3.0.0

**Pull request merged**